### PR TITLE
Various petition changes, grouped by commit

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: basket/petition/tests/test_views.py [hunk 1]
```diff
@@ -54,7 +55,7 @@ def test_petition_post_success(client, mocker):
 
 @pytest.mark.django_db
 def test_petition_post_invalid(client, mocker):
-    mock_send_mail = mocker.patch("basket.petition.views.send_email_confirmation")
+    mock_send_mail = mocker.patch("basket.petition.models.send_email_confirmation")
```
Comment (stevejalim) [inline basket/petition/tests/test_views.py:58]: I was expecting to see `"basket.petition.models.Petition.send_email_confirmation"` - is that patching def right?

File: basket/petition/tests/test_views.py [hunk 2]
```diff
@@ -98,6 +98,27 @@ def test_petition_email_invalid(client, mocker):
...
+        "email": "thedude@example.com",
+        "title": "Dude",
+        "affiliation": "The Knudsens",
+    }
+    response = client.post(url, data=data)
+    assert response.status_code == 200
+    assert response.json() == {
+        "status": "error",
+        "errors": {
+            "__all__": "Database error",
```
Comment (stevejalim) [inline basket/petition/tests/test_views.py:116]: Is this raised because of the emoji? Just wondering how it's triggered.

**Review by stevejalim — APPROVED** [review]
LGTM, with a couple of questions

r+wc

File: basket/petition/tests/test_views.py (hunk 1 above)
Comment (robhudson) [reply basket/petition/tests/test_views.py:58]: Ah, I'm patching the task which the model method calls.

File: basket/petition/tests/test_views.py (hunk 2 above)
Comment (robhudson) [reply basket/petition/tests/test_views.py:116]: Yeah an emoji will trigger it, or most any other database error, but the emoji is an easy test.

**Review by pmac — APPROVED** [review]
🥇

**Pull request merged**